src/cli/one_helper/onezone_helper.rb

```diff
@@ -14,8 +14,373 @@
+    end
+
+    # Process files and folders
+    #
+    # @param sync_db [Boolean] True to sync database
+    def process_files(sync_db)
+        # Files to be copied
+        copy_onedconf
+
+        FILES.each do |file|
+            copy_and_check(file[:name], file[:service])
+        end
+
+        # Folders to be copied
+        FOLDERS.each do |folder|
+            copy_folder(folder[:name], folder[:service])
+        end
+
+        restart_services
+
+        # Sync database
+        sync_db
+    end
+
+    private
+
+    # Get database configuration
+    #
+    # @param configs [Object] Configuration
+    def fetch_db_config(configs)
+        configs.store(:backend, configs[:raw]['/TEMPLATE/DB/BACKEND'])
+
+        if configs[:backend] == 'mysql'
+            configs.store(:server, configs[:raw]['/TEMPLATE/DB/SERVER'])
+            configs.store(:user, configs[:raw]['/TEMPLATE/DB/USER'])
+            configs.store(:password, configs[:raw]['/TEMPLATE/DB/PASSWD'])
+            configs.store(:dbname, configs[:raw]['/TEMPLATE/DB/DB_NAME'])
+            configs.store(:port, configs[:raw]['/TEMPLATE/DB/PORT'])
+            configs[:port] = '3306' if configs[:port] == '0'
+        else
+            STDERR.puts 'No mysql backend configuration found'
+            exit(-1)
+        end
+    end
+
+    # Get federation configuration
+    #
+    # @param configs [Object] Configuration
+    def fetch_fed_config(configs)
+        configs.store(:server_id,
+                        configs[:raw]['/TEMPLATE/FEDERATION/SERVER_ID'])
+        configs.store(:zone_id,
+                        configs[:raw]['/TEMPLATE/FEDERATION/ZONE_ID'])
+    end
+
+    # Replaces a file with the version located on a remote server
+    # Only replaces the file if it's different from the remote one
+    #
+    # @param file    [String] File to check
+    # @param service [String] Service to restart
+    def copy_and_check(file, service)
+        puts "Checking #{file}"
+
+        temp_file = Tempfile.new("#{file}-temp")
+
+        run_command(
+            "scp -i #{@oneadmin_identity_file} #{SSH_OPTIONS} " \
+            "oneadmin@#{@remote_server}:/etc/one/#{file} #{temp_file.path}"
+        )
+
+        if !FileUtils.compare_file(temp_file, "/etc/one/#{file}")
+            FileUtils.cp(temp_file.path, "/etc/one/#{file}")
+
+            puts "#{file} has been replaced by #{@remote_server}:#{file}"
+
+            @opennebula_services[service] = true
+        end
+    ensure
+        temp_file.unlink
+    end
+
+    # Copy folders
+    #
+    # @param folder  [String] Folder to copy
+    # @param service [String] Service to restart
+    def copy_folder(folder, service)
+        puts "Checking #{folder}"
+
+        output = run_command(
+            "rsync -ai\
+            -e \"ssh #{SSH_OPTIONS} -i #{@oneadmin_identity_file}\" " \
+            "oneadmin@#{@remote_server}:/etc/one/#{folder}/ " \
+            "/etc/one/#{folder}/"
```

Comment:
One comment for all the remote calls, I don't think you can rely on the fact that you can connect to oneadmin on another frontend. Also, the accesses are mixed, somewhere the `oneadmin` is enforced, somwhere the current user (e.g., `SSHCommand`) is used. So, ideally it should try the unspecified current user, if it doesn't work, it can try oneadmin, if it doesn't work, it should fail.